contracts/UFragmentsPolicy.sol

```diff
@@ -340,7 +371,28 @@ contract UFragmentsPolicy is Ownable {
             rebaseFunctionUpperPercentage,
             rebaseFunctionGrowth
         );
-        return uFrags.totalSupply().toInt256Safe().mul(rebasePercentage).div(ONE);
+
+        int256 currentSupply = uFrags.totalSupply().toInt256Safe(); // (or) supplyHistory[epoch]
+        int256 newSupply = ONE.add(rebasePercentage).mul(currentSupply).div(ONE);
+
+        // When supply is decreasing:
+        // We limit the supply delta, based on recent supply history.
+        if (rebasePercentage < 0) {
+            int256 maxSupplyInHistory = currentSupply;
+            for (uint8 i = 1; i < epochLookback && epoch > i; i++) {
```

Comment:
This structure seems much easier to read and harder to make mistakes with. Uses epochs directly, rather than offsets.
```suggestion
            for (uint256 e = epoch - epochLookback; e < epoch; e++) {
```

Reply:
Runs into an error when epoch < lookback